src/Microsoft.TemplateEngine.Cli/New3Command.cs

```diff
@@ -281,37 +312,56 @@ private void HandlePostActions(TemplateCreationResult creationResult)
             postActionDispatcher.Process();
         }
 
-        private void DisplayTemplateList(bool showAll = false)
+        // Checks the result of TemplatesToDisplayInfoAbout()
+        // If they all have the same group identity, return them.
+        // Otherwise retun an empty list.
+        private IEnumerable<ITemplateInfo> TemplatesToShowDetailedHelpAbout()
         {
-            IReadOnlyCollection<ITemplateInfo> results = null;
+            IReadOnlyList<ITemplateInfo> candidateTemplates = TemplatesToDisplayInfoAbout();
 
-            try
+            if (candidateTemplates.Count() == 0)
```

Comment:
Since this is an IReadOnlyList, we can use .Count == 0 rather than the extension method